### DISCUSSION TITLE
Multiple root files in Latex VSCode project?

### DISCUSSION THREAD

**Discussion by asapsmc** [James-Yu/LaTeX-Workshop#4140] [Q&A]
I've been using the LateX-Workshop extension in VSCode for a while, so first of all, thanks to the entire community for this excellent tool.

My need right now is that I have to prepare an application where I need to submit a CV, a motivation letter, and a 2-year plan. 
Right now, I'm using multiple workspaces (one per output document), but this is unsatisfying as these documents are all related, and I'd prefer to have this under the same workspace.
Is this possible?
I don't know if this would be the best approach, but, e.g., a single VSCode workspace with multiple root tex files inside so that I can have multiple output pdfs?

**Comment by jlelong** [comment]
LaTeX-Workshop should work prperly with several root documents in the same workspace (or folder). If not, do not hesitate to tell us more about your workflow.

**Reply by asapsmc** [reply]
I'm currently trying to set up a project in LaTeX-Workshop with multiple root documents within the same workspace. However, I've encountered difficulties, as the available documentation primarily addresses scenarios with a single root file. This leaves me uncertain about the best practices for organizing multiple root documents in terms of folder structure and distinct compilation recipes. Could you point me toward any specific resources or examples that detail managing multiple root files in one workspace? Thanks in advance.

**Reply by jlelong** [reply]
You can simply put multiple root files in the same directory and LW will pick the one associated to the active editor following the workflow described at https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#multi-file-projects

Except if you are doing very specific things, you should not need one recipe per root file.

**Reply by josh-hernandez-exe** [reply]
> LaTeX-Workshop should work prperly with several root documents in the same workspace (or folder). If not, do not hesitate to tell us more about your workflow.

I'm actually trying to do a similar thing with a personal resume project. Each root file is a mixture of different page counts and short/long job discriptions, different job histories. I can go to each root file and build it fine. Is there a way to build a recipe that builds all the root files? The multi-file section is more focused on finding the correct root file of all the potential ones, and it isn't clear to me how I can go about building what I want. Though I may have overlooked something from the documentation.

**Reply by jlelong** [reply]
LaTeX-Workshop is not designed to process several root files at a time. If you want to build several files from a single build call,
- you can set `@default_files` in a `.latexmkrc` file and call `latexmk` with no file argument. See https://mg.readthedocs.io/latexmk.html and https://tex.stackexchange.com/a/233796 for instance
- you can use the [external build command mechanism](https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#external-build-command) with your own build script.
